packages/components/nodes/chains/GraphCypherQAChain/GraphCypherQAChain.ts

```diff
@@ -0,0 +1,273 @@
+        const qaModel = nodeData.inputs?.qaModel
+        const graph = nodeData.inputs?.graph
+        const cypherPrompt = nodeData.inputs?.cypherPrompt as BasePromptTemplate | FewShotPromptTemplate
+        const qaPrompt = nodeData.inputs?.qaPrompt as BasePromptTemplate
+        const returnDirect = nodeData.inputs?.returnDirect as boolean
+        const returnIntermediateSteps = nodeData.inputs?.returnIntermediateSteps as boolean
+
+        const omitQaPrompt = cypherPrompt instanceof FewShotPromptTemplate
+
+        // Handle prompt values if they exist
+        let cypherPromptTemplate: PromptTemplate | FewShotPromptTemplate
+        let qaPromptTemplate: PromptTemplate | undefined
+
+        if (cypherPrompt instanceof PromptTemplate) {
+            cypherPromptTemplate = new PromptTemplate({
+                template: cypherPrompt.template as string,
+                inputVariables: cypherPrompt.inputVariables
+            })
+        } else if (cypherPrompt instanceof FewShotPromptTemplate) {
+            const examplePrompt = cypherPrompt.examplePrompt as PromptTemplate
+
+            cypherPromptTemplate = new FewShotPromptTemplate({
+                examples: cypherPrompt.examples,
+                examplePrompt: examplePrompt,
+                inputVariables: cypherPrompt.inputVariables,
+                prefix: cypherPrompt.prefix,
+                suffix: cypherPrompt.suffix,
+                exampleSeparator: cypherPrompt.exampleSeparator,
+                templateFormat: cypherPrompt.templateFormat
+            })
+        } else {
+            cypherPromptTemplate = cypherPrompt as PromptTemplate
+        }
+
+        if (qaPrompt instanceof PromptTemplate) {
+            qaPromptTemplate = new PromptTemplate({
+                template: qaPrompt.template as string,
+                inputVariables: qaPrompt.inputVariables
+            })
+        }
+
+        if (!cypherModel || !qaModel) {
+            if (!model) {
+                throw new Error('Model or is required when Cypher Model or QA Model are not provided')
+            }
+        }
+
+        // validate cypherPromptTemplate is PromptTemplate and qaPromptTemplate must be exists
+        if (cypherPromptTemplate instanceof PromptTemplate && !qaPromptTemplate) {
+            throw new Error('QA Prompt is required when Cypher Prompt is a Prompt Template')
+        }
+
+        // Validate required variables in prompts
+        if (!cypherPromptTemplate.inputVariables.includes('schema') || !cypherPromptTemplate.inputVariables.includes('question')) {
+            throw new Error('Cypher Generation Prompt must include {schema} and {question} variables')
+        }
+
+        const chain = GraphCypherQAChain.fromLLM({
+            llm: model,
+            qaLLM: qaModel,
+            cypherLLM: cypherModel,
+            graph,
+            cypherPrompt: cypherPromptTemplate,
+            qaPrompt: omitQaPrompt ? undefined : qaPromptTemplate,
```

Comment:
why do we need `omitQaPrompt` ? cant we just pass in `qaPrompt: qaPromptTemplate`?

Reply:
omitQaPrompt variable removed